CodeGeneration/Sources/SyntaxSupport/Node.swift

```diff
@@ -85,6 +85,70 @@ public class Node {
     }
   }
 
+  /// Get the first potential token of a node recursively
+  /// This information is used by incremental parse
+  public var firstPotentialToken: [TokenChoice] {
+    let getChildTokenChoice: (Child) -> [TokenChoice] = { child in
```

Comment:
Maybe we could move this recursion logic to `IncrementalParseUtilsFile.swift`. We only produce `[TokenChoice]` for child whose kind is `.token` and make `.collection` and `.nodeChoices` in the form of `SyntaxKind.someKind.firstPotentialToken + SyntaxKind.anotherKind.firstPotentialToken + ...` in `IncrementalParseUtils.swift`. 

This should reduce the code size. What do you think?